main.py

```diff
@@ -24,62 +26,78 @@
 
 
 class Item:
-    def __init__(self, title, price, torg, url, date, image):
+    def __init__(self, title, price, torg, url, image):
         self.title = title
         self.price = price
         self.torg = torg
         self.url = 'https://www.ebay-kleinanzeigen.de' + url
-        self.date = date
+        #self.date = date
         self.image = image
 
     def __repr__(self):
-        return f'{self.title} - {self.price} - {self.date}'
+        return f'{self.title} - {self.price}'
 
     def __str__(self):
         result = f'{self.title} - {self.price}'
         if self.torg:
             result += ' VB'
-        result += f'\n\t{self.date}\n'
+        #result += f'\n\t{self.date}\n'
 
         result += self.url
         result += '\n'
         return result
 
 
 def get_items_per_url(url):
-    qq = requests.get(url)
+    log = utils.get_logger()
+    # Simulate browser headers
+    headers = {
+        'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_1) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/39.0.2171.95 Safari/537.36',
+        'Host': 'www.ebay-kleinanzeigen.de',
+        'Accept': '*/*',
+        }
+
+    qq = requests.get(url, headers=headers)
 
     text = qq.text
-    # with open(r'C:\Users\Kleinanzeigen\Desktop\qq.html', 'r') as f:
-    #     text = f.read()
+    
 
-    articles = re.findall('<article(.*?)</article', text, re.S)
 
+    articles = re.findall('<article(.*?)</article', text, re.S)
+    log.info('Articles length %s' % len(articles))
     items = []
     for item in articles:
-        if results := re.findall('<a.*?href="(.*?)">(.*?)</a>', item, re.S):
-            url, name = results[0]
+        soup = BeautifulSoup(item, 'html.parser')
+        soup_result = soup.find_all("a", {"class": 'ellipsis'})
+        if len(soup_result) > 0:
+            url = soup_result[0]['href']
+            name = soup_result[0].text
         else:
             continue
 
-        price_line = re.findall('<strong>(.*?)</strong>', item, re.S)[0]
+        price_line = re.findall('aditem-main--middle--price">(.*?)</p>', item, re.S)
+        if len(price_line) > 0:
+            price_line = price_line[0]
+        else:
+            price_line = 0
```

Comment:
I think this will crash as regexp is applied to `price_line` at line 85, it cannot be `int`.

Reply:
I've changed it to a string, that should work